pkg/amber/provenance.go

```diff
@@ -100,6 +100,25 @@ func (p *ValidatedProvenance) GetBuildCmd() ([]string, error) {
 	return buildConfig.Command, nil
 }
 
+// GetConfigSourceURI returns the URI of the configuration.
+func (p *ValidatedProvenance) GetConfigSourceURI() string {
```

Comment:
The same values can be retrieved the same way from slsav02 provenances too. Is that right?
Should not that we be done in this PR?

Reply:
Now that the `ValidatedProvenance` types are unified, I think this can be moved to and be used both in `fromAmber` and `fromSlsav02`.

Reply:
I don't think `GetConfigSourceURI` and `GetMaterialsGitURI` can be moved to `types`. 

We need to know which predicate it is, i.e., we would need to add a dependency on slsa in `types`. Or am I missing something?

Reply:
Not to `types` but to `slsav02`. Because currently there is no `GetConfigSourceURI` for `slsav02`.

Reply:
Do you think I should add `GetConfigSourceURI` and `GetMaterialsGitURI` here: https://github.com/project-oak/transparent-release/blob/main/pkg/intoto/slsa_provenance/v0.2/provenance.go ?

For me, I mainly need to be able to test, and as our current test provenance is amber, I added the implementations to the amber pkg.

Reply:
Yes. https://github.com/project-oak/transparent-release/blob/main/pkg/intoto/slsa_provenance/v0.2/provenance.go  is a better place for those new functions. 

Are `GetConfigSourceURI` and `GetMaterialsGitURI` covered in any of the tests? I don't see them.

> and as our current test provenance is amber, I added the implementations to the amber pkg.

Do you mean we don't have any tests for `slsav02`? Since we have the code for it, we should have tests for it too. Actually, `slsav02` is more important than `amber`, as `amber` is deprecated.

Reply:
I meant [`testdata/provenance.json`](https://github.com/project-oak/transparent-release/blob/main/testdata/provenance.json).

As far as I can tell, the tests rely a lot on `testdata/provenance.json`, but I don't think we can change this easily.

I am not sure how I could generate a `testdata/provenance.json` for slsa.v02.

Do you have an idea here?

Reply:
We need a new testdata file for slsav02 provenance. You can use [this example](https://paste.googleplex.com/5755513575833600). Perhaps rename `testdata/provenance.json` to `testdata/amber_provenance.json`.

Reply:
Actually, why are we using `testdata/provenance.json` and not `schema/provenance/v1/example.json`? Some tests use the latter.

Reply:
Yes, I'll do that in #197 , too, I think there it fits.

Reply:
Thanks. I had a closer look, and I think perhaps `schema/provenance/v1/example.json` can be removed. But let's discuss that separately.